runtime/parachains/src/hrmp/benchmarking.rs

```diff
@@ -0,0 +1,294 @@
+		recipient,
+		capacity,
+		message_size
+	));
+
+	if matches!(until, ParachainSetupStep::Requested) {
+		return output
+	}
+
+	assert_ok!(Hrmp::<T>::hrmp_accept_open_channel(recipient_origin.into(), sender));
+	if matches!(until, ParachainSetupStep::Accepted) {
+		return output
+	}
+
+	Hrmp::<T>::process_hrmp_open_channel_requests(&Configuration::<T>::config());
+	if matches!(until, ParachainSetupStep::Established) {
+		return output
+	}
+
+	let channel_id = HrmpChannelId { sender, recipient };
+	assert_ok!(Hrmp::<T>::hrmp_close_channel(sender_origin.clone().into(), channel_id));
+	if matches!(until, ParachainSetupStep::CloseRequested) {
+		// NOTE: this is just for expressiveness, otherwise the if-statement is 100% useless.
+		return output
+	}
+
+	output
+}
+
+const PREFIX_0: u32 = 10_000;
+const PREFIX_1: u32 = PREFIX_0 * 2;
+const MAX_UNIQUE_CHANNELS: u32 = 128;
+
+static_assertions::const_assert!(MAX_UNIQUE_CHANNELS < PREFIX_0);
+
+frame_benchmarking::benchmarks! {
+	where_clause { where <T as frame_system::Config>::Origin: From<crate::Origin> }
+
+	hrmp_init_open_channel {
+		let sender_id: ParaId = 1u32.into();
+		let sender_origin: crate::Origin = 1u32.into();
+
+		let recipient_id: ParaId = 2u32.into();
+
+		// make sure para is registered, and has enough balance.
+		let deposit: BalanceOf<T> = Configuration::<T>::config().hrmp_sender_deposit.unique_saturated_into();
+		register_parachain_with_balance::<T>(sender_id, deposit);
+		register_parachain_with_balance::<T>(recipient_id, deposit);
+
+		let capacity = Configuration::<T>::config().hrmp_channel_max_capacity;
+		let message_size = Configuration::<T>::config().hrmp_channel_max_message_size;
+	}: _(sender_origin, recipient_id, capacity, message_size)
+	verify {
+		assert_last_event::<T>(
+			Event::<T>::OpenChannelRequested(sender_id, recipient_id, capacity, message_size).into()
+		);
+	}
+
+	hrmp_accept_open_channel {
+		let [(sender, _), (recipient, recipient_origin)] =
+			establish_para_connection::<T>(1, 2, ParachainSetupStep::Requested);
+	}: _(recipient_origin, sender)
+	verify {
+		assert_last_event::<T>(Event::<T>::OpenChannelAccepted(sender, recipient).into());
+	}
+
+	hrmp_close_channel {
+		let [(sender, sender_origin), (recipient, _)] =
+			establish_para_connection::<T>(1, 2, ParachainSetupStep::Established);
+		let channel_id = HrmpChannelId { sender, recipient };
+	}: _(sender_origin, channel_id.clone())
+	verify {
+		assert_last_event::<T>(Event::<T>::ChannelClosed(sender, channel_id).into());
+	}
+
+	// NOTE: a single parachain should have the maximum number of allowed ingress and egress
+	// channels.
+	force_clean_hrmp {
+		// ingress channels to a single leaving parachain that need to be closed.
+		let i in 0 .. (<T as configuration::Config>::HrmpMaxInboundChannelsBound::get() - 1);
+		// egress channels to a single leaving parachain that need to be closed.
+		let e in 0 .. (<T as configuration::Config>::HrmpMaxOutboundChannelsBound::get() - 1);
```

Comment:
Just out of my inexperience: why do we check it for every variable and just not once for the bound const variable?

Reply:
(if I have understood your question correctly:) We want the benchmarking machinery to tell us the dependency on `i` and `e`, for which you need to run the benchmark in a range of values for both `i` and `e`. The final outcome of this would be `fn(i, e) -> weight`.

Reply:
But why do we want us to tell the dependency on those? `i` and `e` are constants anyway?

Reply:
their upper bound is constant and we happen to use that here as well, If you see where `T::WeightInfo::force_clean_hrmp` is being called, these two parameters need to be provided (similar to the other comment, it is a root call and we expect root to give a proper estimate of the witness data)